### DISCUSSION TITLE
403 error (not antivirus problem)

### DISCUSSION THREAD

**Discussion by kenlbro** [aclap-dev/video-downloadhelper#137] [Help and Q&A] [answered]
### Your question

I use video downloadhelper to help me with my online college courses (capture lectures and such).  I used to be able to download lectures using previous versions of video-downloadhelper, but with the "default downloader" being changed, that is no longer able to get past cloudflare, and I get an access denied message instead.

I am guessing this has to do with not downloading using the alternate tool.  I did a test with an older version of video-downloadhelper, and that holds true.  How can I get access to the other download protocol under the latest version?

Here is an example of the options available under an older version (which works).
![successful](https://github.com/aclap-dev/video-downloadhelper/assets/116030153/670c95a7-9399-4a4b-a015-7ed60c5a1268)
In this image, the option that says "Lesson 3 - 29:12 - MP4" will download without error.  But with the latest version of video-downloadhelper, this is not even an option to select, and will not download.

These are the options available under the current version (which are blocked by cloudflare):
![failure_cloudflare_blocking](https://github.com/aclap-dev/video-downloadhelper/assets/116030153/60258fe2-2ebc-4300-8b15-dd2b7722f6b2)
Notice how the older option that works is not longer listed as an option.

Thank you so much for any direction here.

### Addon & CoApp version and details

This is the version which broke my download capability (the most recent version).  I was using newer versions just last week that worked fine, but autoupdate removed it, so I cannot show you that, can only go to a really old version.

This is the version that is "broken" (or I'm not understanding how to use the older download method):
version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "/home/kbrobst/vdhcoapp-2.0.10/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/kbrobst",
  "converterBinary": "/home/kbrobst/vdhcoapp-2.0.10/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}

This is the comparison, really old, but my only not-auto-updated version, and it still works with the same site:

version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "/home/kbrobst/vdhcoapp-2.0.10/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/kbrobst",
  "converterBinary": "/home/kbrobst/vdhcoapp-2.0.10/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}

**Comment by kenlbro** [comment]
Sorry, paste error.  The older version that works is this (although the prior version before 8.2.0.23 also functioned):

Video DownloadHelper 8.0.0.6
Video DownloadHelper
Version 8.0.0.6
Browser locale: en-US
Production build
Built on Tue Aug 22 2023 18:17:53 GMT+0200 (Central European Summer Time)
Build options: browser=firefox
⊖
Platform
Platform Linux x86-64
Browser Mozilla Firefox 88.0
⊖
Companion App outdated - please update
Found companion app: VdhCoApp 1.3.0
Latest available version is 1.6.2
Companion app binary: /opt/net.downloadhelper.coapp/bin/net.downloadhelper.coapp-linux-64

**Comment by paulrouget** [comment]
Can you copy paste the access denied error?

**Reply by kenlbro** [reply]
Sure, here is a sample:

SideDownload error:

https://d12eajhnrzqg7n.cloudfront.net/0b023723-b1a8-4d2a-9bd2-302063343fa0_screen_capture_video_1675798362_display0.mp4_hls/720p/0b023723-b1a8-4d2a-9bd2-302063343fa0_screen_capture_video_1675798362_display0.mp4_hls.m3u8: Server returned 403 Forbidden (access denied)

**Comment by paulrouget** [comment]
- Can you update to CoApp 2.0.17: https://www.downloadhelper.net/w/CoApp-Installation
- And please share the link you're downloading from.

(Also make sure you don't have anything that blocks the coapp binary to do tcp connections? like… a firewall of some sort - but I imagine you don't have such thing as you're on Linux)

**Reply by kenlbro** [reply]
I updated to CoApp 2.0.17, went back to the original lecture video and attempted to download it.

The URL for the video shows as this: https://msudenver.yuja.com/V/Video?v=4302742&node=14518896&a=1032262446&autoplay=1

(access to this content requires login, so not sure if it gives anything to test against):

-----
I get a similar error trying to download the lecture:

SideDownload error:

[in#0 @ 0x591dac0] Error opening input: Server returned 403 Forbidden (access denied)
Error opening input file https://d12eajhnrzqg7n.cloudfront.net/0b023723-b1a8-4d2a-9bd2-302063343fa0_screen_capture_video_1643648833_display0.mp4_hls/720p/0b023723-b1a8-4d2a-9bd2-302063343fa0_screen_capture_video_1643648833_display0.mp4_hls.m3u8.
Error opening input files: Server returned 403 Forbidden (access denied)

-----
Here's my updated configuration:

version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.17",
  "binary": "/opt/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/kbrobst",
  "converterBinary": "/opt/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "63b936e"
}

**Reply by paulrouget** [reply]
I can reproduce. Thank you. I'll investigate.

**Comment by GrampaWildWilly** [comment]
I must say I'm surprised your URL let me in.  When you said it was an education course, I figured I would hit a prompt for login & password.  It wouldn't surprise me if your URL stops working in a while.  While.  There's a specific time for you.  While.  An hour?  Six hours?  Who knows?  These things are quite often on some kind of timeout.  Whatever . . .

This is one of the oddest cases I've encountered.  Talk about separate video & audio.  I also got an error trying to download this with VDH.  So I went to our tried & true Plan B.  I opened the Firefox Network Monitor & reloaded the page.  It showed me 3 HLS manifests, as shown in the attached screenshot.  The third one was for a stream so it was of no interest.  But the first 2 were both HLS master manifests.  In the image, the second one is highlighted, with the first one immediately above it.

As usual, I ran the 2 manifests through ffprobe.  I was thinking ahead of time that they were 2 identical manifests resulting from my reloading the page.  That's actually false since a page reload also clears the Network Monitor, so there wouldn't be leftovers from a previous display of the page.  But I wasn't totally with it.  In any case, the 2 ffprobes on those 2 master manifests show something really weird.  The first manifest shows only 1 variant & it is for an audio track.  Just audio.  And only one of them.  Similarly, the second manifest shows only 1 variant & it is for a video track.  Just video.  And only one of them.  Can you say separate video & audio?  You can see this in the attached ffprobe report, which contains the output from the 2 ffprobe invocations.

But the duration of the tracks is the same.  OK.  Awfully strange way of presenting media.  The usual thing would be to include the video & audio as tracks within a single master manifest.  That's the way most web sites do things.  This one is unique in my experience.  But whatever.

I downloaded the 2 tracks in separate ffmpeg sessions, shown in the 2 attached ffmpeg logs.  I was a bit worried when I saw that the frame rate for the video is 5fps.  But I played the video & audio tracks using VLC's synchronous playback feature.  It turns out the video is just a PowerPoiut slide show, hence the ridiculous frame rate.  It really is 5fps & there's no problem with it.  I also had audio but I must say it was recorded at an awfully low level.  I had to crank the volume up way past what is usually uncomfortable just to hear her.  And even then, I could hardly understand her.  I'm not sure if she has an accent or she just has a bad habit of swallowing her words & speaking too fast.  I found her nearly impossible to understand.  Not that I tried very hard.  I didn't sit & watch this all the way through, just sampled it at intervals to make sure it was all there, including about the last 20 seconds, just to verify that the audio went right to the end.

So this is a most strange case.  Seems like it ought to be fairly simple to resolve but it is really weird.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/e0918c18-5cf9-4af4-8eb2-64b31bd72608)
[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/14317872/ffprobe.txt)
[Video mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14317874/Video.mp4.Log.txt)
[Audio mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14317875/Audio.mp4.Log.txt)

**Reply by paulrouget** [reply]
I'm think this might be a HTTP headers issue. I can only see that really…

**Reply by GrampaWildWilly** [reply]
I can only nod my head in uncomprehending wonder at that & assume you know what that means because I sure don't.

In any case, after I did all of that with ffmpeg, it occurred to me that I ought to try again with VDH.  The VDH menu was offering me 3 variants.  I tried each one in turn.  The first one got the error.  VDH was perfectly happy to download the second & third variants.  But they both turned out to be the audio track without the video.  They both played fine, except for the volume thing & her being incomprehensible, but they were both just the audio track.

**Reply by kenlbro** [reply]
In the older versions of videodownloadhelper, it too would provide me with the audio in one stream and the video in another.  I just worked around that.  But with this current version, I cannot download the video stream.

In other lectures, it is more than a powerpoint, you just happened to see one of the less interactive sessions.  And, yes, I'm guessing those URL's will go inactive after a while.

Thanks again for looking.

**Reply by GrampaWildWilly** [reply]
Oh.  Forgot to mention I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3155, Firefox 122.0.1 64-bit, licensed VDH 8.2.0.27a1 beta, CoApp 2.0.17.  Plus ffprobe/ffmpeg versions you can see in the logs.

**Reply by GrampaWildWilly** [reply]
I lied but that's Paul's fault.  I now have VDH 8.2.1.0a1 beta.  Instead of 3 variants on the VDH menu, I'm now seeing 2.  The first one still gets the error; the second one is the audio track without video.

**Comment by paulrouget** [comment, marked as answer]
Ok -  I figured it out.

This is another case of Http Headers not being standard.

Here is a way around that:

In the settings, click "More", then "Video Qualities", then uncheck "Ignore Low Video Qualities".

Then that page will show more options to download from, and the MP4 one will work.

**Reply by GrampaWildWilly** [reply]
I've always had that option unchecked.

I revisited the link @kenlbro posted on February 16 above.  Amazingly, the link still works.  VDH offered me 4 variants.  The first one gave an error.  The second one downloaded the audio without video.  The third one downloaded the video without audio.  The fourth one downloaded the audio without video, a duplicate of the second one.  This is at least something of an improvement in that one of the variants now gives me the video, which wasn't true the other day.  Once you've got the video & audio in separate files, you have a choice.  You can play the 2 files synchronously in VLC using its standard features.  Or you can merge the video & audio using the VDH tool.  Your choice.  But this isn't 100% cured.  On the other hand, you say this one is non-standard.  Does that mean this is the best we can expect?  I know if it were up to me, I'd live with it.  But I'm not the one who reported this.

Since the other day, my environment has changed: Windows 11 Pro 64-bit version 23H2 build 22631.3155, Firefox 123.0 64-bit, licensed VDH 8.2.1.3a1 beta, CoApp 2.0.19.

**Reply by kenlbro** [reply]
This was exactly what I needed.  Thank you!

**Reply by Tureto** [reply]
I had this issue and I solved:
In the settings, click "More", then "Video Qualities", then check "Show Low Video Qualities".
And select RAWHLS video on download menu
![image](https://github.com/user-attachments/assets/a5e4e196-5651-4ed8-8b69-9253ad01c54d)
